odbc/src/api/functions.rs

```diff
@@ -2013,8 +2004,10 @@ unsafe fn sql_get_infow_helper(
                 | InfoType::SQL_CONVERT_VARBINARY
                 | InfoType::SQL_CONVERT_LONGVARBINARY
                 | InfoType::SQL_CONVERT_GUID => {
-                    // MongoSQL does not support CONVERT.
-                    i16_len::set_output_fixed_data(&SQL_U32_ZERO, info_value_ptr, string_length_ptr)
+                    // MongoSQL does not support the CONVERT scalar function, but clients also use
+                    // this to apply the CAST syntactic construct. The value we return for
+                    // SQL_CONVERT_FUNCTIONS alerts the client that we expect CAST and not CONVERT.
+                    i16_len::set_output_fixed_data(&MONGO_CAST_SUPPORT, info_value_ptr, string_length_ptr)
```

Comment:
I wonder if we should advertise cast which are not supported since the data will always be Null.
Things like BOOL to TIMESTAMP or TIMESTAMP to INT for example.
Or is it better since then it will fold no matter what?

Reply:
We decided to reevaluate this during Direct Query